### DISCUSSION TITLE
Nesting of non nested symbols for Scala's new Optional Braces syntax in Symbols

### DISCUSSION THREAD

**Discussion by achinaou** [github-linguist/linguist#7115] [Classification]
Scala 3 has a [new syntax feature](https://docs.scala-lang.org/scala3/reference/other-new-features/indentation.html#optional-braces-2).

When used, the structures defined at the same level in the code appear as nested in the Symbols panel.

Provided code to reproduce the issue:

```scala
object Application:

  val name    = "Application"
  val version = "0.0.1"

  def logic: Unit =
    println(s"The current version of the $name is $version")

  def main(args: Array[String]): Unit =
    logic
```

![Screenshot from 2024-10-26 14-34-26](https://github.com/user-attachments/assets/3835f8db-5e9b-4ee6-a0b7-1dcd17292ec4)

**Comment by lildude** [comment]
This has nothing to do with Linguist. Please use the "Contact" link at the bottom of a page showing the issue to report this to GitHub support.

**Reply by achinaou** [reply]
Hello, Colin, and thanks for the response!

You are correct that this probably isn't the right place for the issue/discussion.

I misunderstood the suggestion made by GitHub support, which I contacted before opening this discussion.
They suggested opening an issue in the **grammar** that the `linguist` uses rather than in the `linguist` itself.
Which, if I'm not mistaken, for Scala, it's [this one](https://github.com/scala/vscode-scala-syntax), right?

**Reply by achinaou** [reply]
I'm just trying to find the place that's the most appropriate one for this issue to be reported.

**Reply by lildude** [reply]
The symbol tree on the right isn't influenced by the grammar we use for syntax highlighting so I'm not sending you over to the grammar repo. I've done a bit of digging, found your original support ticket, and I've opened an issue for this with the team responsible for this functionality and added an internal note to your support ticket.

In short, this is on GitHub to resolve and there's nothing for you to do. Thanks for the ping and sorry for the bounce-around.

**Reply by achinaou** [reply]
Thank you very much, Colin!
